### PR TITLE
Add installation_source field to the host telemetry

### PR DESCRIPTION
**Pull request by arbulu89**
Add the installation_source field to the host telemetry model.

This field is not required in the agent side, we so it is still backwards compatible with older agent version. In this cases, it is populated with `Unknown` string.

Depends on: https://github.com/trento-project/agent/pull/64

### CODE REVIEW

**Review by rtorrero — APPROVED** [review]
I had imaged here he would use something like the repo url to identify the installation source, but I guess that what you propose here is more flexible. Looks good to me @arbulu89, thanks!

**Review by fabriziosestito — CHANGES_REQUESTED** [review]

File: lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex [hunk 1]
```diff
@@ -16,5 +25,6 @@ defmodule Trento.Integration.Discovery.HostDiscoveryPayload do
     field :total_memory_mb, :integer
     field :socket_count, :integer
     field :os_version, :string
+    field :installation_source, :string, default: "Unknown"
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex]: Done in: https://github.com/trento-project/web/pull/728/commits/6f6c1c750e44b143c890ce858a0ca94cdb2f9606

File: lib/trento/application/read_models/host_telemetry_read_model.ex [hunk 2]
```diff
@@ -18,6 +18,7 @@ defmodule Trento.HostTelemetryReadModel do
     field :socket_count, :integer
     field :total_memory_mb, :integer
     field :sles_version, :string
+    field :installation_source, :string
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/application/read_models/host_telemetry_read_model.ex]: Done: https://github.com/trento-project/web/pull/728/commits/6f6c1c750e44b143c890ce858a0ca94cdb2f9606

**Comment by arbulu89** [comment]
Moving to `Draft` as this change creates a risky tech debt as the event store is not properly upcasted

**Comment by fabriziosestito** [comment]
Unblocked by #751

**Comment by arbulu89** [comment]
Upcasting applied. Waiting for https://github.com/trento-project/web/pull/751 to be rebased

**Comment by arbulu89** [comment]
Hey @fabriziosestito ,
The upcasting implemented https://github.com/trento-project/web/pull/728/commits/ea9ddb932819a6985cf55e5653229ce19ca6b73b. Finally I have added a couple of basic tests, as they are quick to write.

File: lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex [hunk 3]
```diff
@@ -16,5 +25,24 @@ defmodule Trento.Integration.Discovery.HostDiscoveryPayload do
...
+
+    field :installation_source, Ecto.Enum,
+      values: [:community, :suse, :unknown],
+      default: :unknown
+  end
+
+  def changeset(host, attrs) do
+    modified_attrs =
+      attrs
+      |> installation_source_to_downcase
```
Comment (fabriziosestito) [inline lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex:37]: `installation_source_to_downcase(attrs)`

File: lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex [hunk 4]
```diff
@@ -3,7 +3,16 @@ defmodule Trento.Integration.Discovery.HostDiscoveryPayload do
...
+  @required_fields [
+    :hostname,
+    :ip_addresses,
+    :ssh_address,
+    :agent_version,
+    :cpu_count,
+    :total_memory_mb,
+    :socket_count,
+    :os_version
+  ]
```
Comment (fabriziosestito) [inline lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex:15]: just curious, is this needed also we the default? isn't `:all` enough?

File: lib/trento/domain/host/commands/register_host.ex [hunk 5]
```diff
@@ -17,5 +17,9 @@ defmodule Trento.Domain.Commands.RegisterHost do
     field :total_memory_mb, :integer
     field :socket_count, :integer
     field :os_version, :string, default: "Unknown"
+
+    field :installation_source, Ecto.Enum,
+      values: [:community, :suse, :unknown],
+      default: :unknown
```
Comment (fabriziosestito) [inline lib/trento/domain/host/commands/register_host.ex:23]: is the default needed here in the command? i find it a bit opaque that we can build a command without that field and having a default value in place. this is already managed by the payload parsing, so I would not use any default when building the command.

File: lib/trento/domain/host/events/host_details_updated.ex [hunk 6]
```diff
@@ -15,5 +15,11 @@ defmodule Trento.Domain.Events.HostDetailsUpdated do
     field :total_memory_mb, :integer
     field :socket_count, :integer
     field :os_version, :string
+
+    field :installation_source, Ecto.Enum,
+      values: [:community, :suse, :unknown],
+      default: :unknown
```
Comment (fabriziosestito) [inline lib/trento/domain/host/events/host_details_updated.ex:21]: the default is not needed in the event

File: lib/trento/domain/host/events/host_registered.ex [hunk 7]
```diff
@@ -15,6 +15,13 @@ defmodule Trento.Domain.Events.HostRegistered do
     field :total_memory_mb, :integer
     field :socket_count, :integer
     field :os_version, :string
+
+    field :installation_source, Ecto.Enum,
+      values: [:community, :suse, :unknown],
+      default: :unknown
```
Comment (fabriziosestito) [inline lib/trento/domain/host/events/host_registered.ex:21]: same, default is not needed

File: test/support/factory.ex [hunk 8]
```diff
@@ -64,6 +64,7 @@ defmodule Trento.Factory do
       total_memory_mb: Enum.random(1..128),
       socket_count: Enum.random(1..16),
       os_version: Faker.App.semver(),
+      installation_source: Enum.random([:community, :suse, :unknown]),
```
Comment (fabriziosestito) [inline test/support/factory.ex:67]: great!

File: test/trento/domain/host/host_test.exs [hunk 9]
```diff
@@ -522,4 +531,26 @@ defmodule Trento.HostTest do
...
+
+    test "should upcast HostDetailsUpdated event properly" do
+      assert %HostDetailsUpdated{
+               version: 2,
+               installation_source: :unknown
+             } ==
+               %{}
+               |> HostDetailsUpdated.upcast(%{})
+               |> HostDetailsUpdated.new!()
+    end
```
Comment (fabriziosestito) [inline test/trento/domain/host/host_test.exs:554]: I would split these tests into two different files with the event module name to reflect the module structure (e.g. domain/hosts/events/host_details_updated.exs`

Also, the test is a bit unclear, we need to upcast a map that is at version 1, not an empty map because this is what we get from the event store when dealing with old events.

Something like 
```elixir
%{
                 "version" => 1,
# insert all the host details updated events fields that are present in version 1 possibly with a faker
....
               }
```

**Review by fabriziosestito — CHANGES_REQUESTED** [review]
Some minor stuff then good to go

File: lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex (hunk 4 above)
Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex:15]: The `installation_source` field is not mandatory, as old versions of the agent won't send it, and we need to cover this use case

Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/host_discovery_payload.ex:15]: You are right, having a `default` value already fills the value, so `:all` works perfectly fine. I will revert it

File: test/trento/domain/host/host_test.exs (hunk 9 above)
Comment (fabriziosestito) [reply test/trento/domain/host/host_test.exs:554]: or event without the version field, since it should default it at version 1

**Comment by arbulu89** [comment]
> Some minor stuff then good to go

Thanks @fabriziosestito ,
Changes done!

**Review by fabriziosestito — APPROVED** [review]

**Comment by fabriziosestito** [comment]
> > Some minor stuff then good to go
> 
> Thanks @fabriziosestito , Changes done!

@arbulu89 go!
